src/aosm/azext_aosm/cli_handlers/onboarding_core_vnf_handler.py

```diff
@@ -0,0 +1,211 @@
+                    "Created list of artifacts from %s arm template(s) provided: %s",
+                    len(self.config.arm_templates),
+                    acr_artifact_list,
+                )
+            elif isinstance(processor, VHDProcessor):
+                sa_artifact_list = processor.get_artifact_manifest_list()
+                logger.debug(
+                    "Created list of artifacts from vhd image provided: %s",
+                    sa_artifact_list,
+                )
+
+        # Build manifest bicep contents, with j2 template
+        template_path = get_template_path(
+            VNF_TEMPLATE_FOLDER_NAME, VNF_MANIFEST_TEMPLATE_FILENAME
+        )
+        params = {
+            "acr_artifacts": acr_artifact_list,
+            "sa_artifacts": sa_artifact_list,
+        }
+        bicep_contents = render_bicep_contents_from_j2(
+            template_path, params
+        )
+
+        # Create Bicep element with manifest contents
+        bicep_file = BicepDefinitionElementBuilder(
+            Path(VNF_OUTPUT_FOLDER_FILENAME, MANIFEST_FOLDER_NAME),
+            bicep_contents,
+        )
+
+        logger.info("Created artifact manifest bicep element")
+        return bicep_file
+
+    def build_all_parameters_json(self) -> JSONDefinitionElementBuilder:
+        """Create object for all_parameters.json."""
+        params_content = {
+            "location": self.config.location,
+            "publisherName": self.config.publisher_name,
+            "publisherResourceGroupName": self.config.publisher_resource_group_name,
+            "acrArtifactStoreName": self.config.acr_artifact_store_name,
+            "saArtifactStoreName": self.config.blob_artifact_store_name,
+            "acrManifestName": self.config.acr_artifact_store_name + "-manifest",
+            "saManifestName": self.config.blob_artifact_store_name + "-manifest",
+            "nfDefinitionGroup": self.config.nf_name,
+            "nfDefinitionVersion": self.config.version
+        }
+        base_file = JSONDefinitionElementBuilder(
+            Path(VNF_OUTPUT_FOLDER_FILENAME), json.dumps(params_content, indent=4)
+        )
+        return base_file
+
+    def _get_default_config(self, vhd) -> Dict[str, Any]:
+        """Get default VHD config for Azure Core VNF."""
+        default_config = {}
+        if vhd.image_disk_size_GB:
+            default_config.update({"image_disk_size_GB": vhd.image_disk_size_GB})
+        if vhd.image_hyper_v_generation:
+            default_config.update(
+                {"image_hyper_v_generation": vhd.image_hyper_v_generation}
+            )
+        else:
+            # Default to V1 if not specified
+            default_config.update({"image_hyper_v_generation": "V1"})
+        if vhd.image_api_version:
+            default_config.update({"image_api_version": vhd.image_api_version})
+        return default_config
+
+    def _generate_type_specific_nf_application(self, processor) -> "tuple[list, list]":
```

Comment:
Almost identical to Nexus handler. Can you generalise the method and put it in a parent class to deduplicate?